atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java

```diff
@@ -723,28 +727,6 @@ public void setEntryValidator(final EntryValidator validator) {
     this.entryValidator = validator;
   }
 
-  /**
-   * Returns the last applied index.
-   *
-   * @return the last applied index
-   */
-  public long getLastApplied() {
-    return lastApplied;
-  }
-
-  /**
-   * Sets the last applied index.
-   *
-   * @param lastApplied the last applied index and the last applied term
-   */
-  private void setLastApplied(final long lastApplied) {
-    this.lastApplied = Math.max(this.lastApplied, lastApplied);
```

Comment:
Do you think we need the max ? So max of current and previous commit? Idk why this was done before. I had hoped that we have the guarantee that this is increasing :sweat_smile:

Reply:
This is removed code :)

Reply:
Yeah I know but was not sure whether we need this in the new code? :sweat_smile:

Reply:
Ah.. Sorry. I misread it :smile: It is already checked in `setCommitIndex` https://github.com/zeebe-io/zeebe/blob/2a5a970ec48e97f54aee475ae32757c240dcd03e/atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java#L346

Reply:
Ah ok this is still a bit different but ok thanks